semanticz-connector-graphdb/src/main/java/zone/cogni/semanticz/connectors/graphdb/GraphDBSparqlService.java

```diff
@@ -19,107 +19,225 @@
-        .newBuilder()
-        .POST(BodyPublishers.ofString(ttl, StandardCharsets.UTF_8))
-        .header(CONTENT_TYPE, Lang.TURTLE.getHeaderString())
-        .uri(URI.create(config.getSparqlUpdateEndpoint()+"?context=" + URLEncoder.encode("<"+graphUri+">", StandardCharsets.UTF_8)))
-        .build();
-    execute(request, httpClient);
-  }
-
-  @Override
-  public Model executeConstructQuery(String query) {
-    try (QueryExecution queryExecution = getQueryExecutionBuilder().query(query).build()) {
-      return queryExecution.execConstruct();
+
+    @Override
+    public boolean executeAskQuery(String askQuery) {
+        try (QueryExecution queryExecution = QueryExecutionHTTP.service(config.getSparqlEndpoint())
+                .queryString(askQuery)
+                .httpClient(getHttpClient())
+                .build()) {
+            return queryExecution.execAsk();
+        }
     }
-  }
-
-  @Override
-  public void executeUpdateQuery(String updateQuery) {
-    final HttpRequest request = HttpRequest
-        .newBuilder(URI.create(config.getSparqlUpdateEndpoint()))
-        .POST(BodyPublishers.ofString("update=" + URLEncoder.encode(updateQuery, StandardCharsets.UTF_8), StandardCharsets.UTF_8))
-        .header(CONTENT_TYPE, Constants.APPLICATION_FORM_URLENCODED_VALUE)
-        .build();
-    execute(request, httpClient);
-  }
-
-  @Override
-  public <R> R executeSelectQuery(String query, Function<ResultSet, R> resultHandler) {
-    try (QueryExecution queryExecution = getQueryExecutionBuilder().query(query).build()) {
-      return resultHandler.apply(queryExecution.execSelect());
+
+
+    @Override
+    public void dropGraph(String graphUri) {
+        executeUpdateQuery("clear graph <" + graphUri + ">");
+    }
+
+    @Override
+    public void updateGraph(String graphUri, Model model) {
+        Objects.requireNonNull(graphUri, "graphUri must not be null");
+        Objects.requireNonNull(model, "model must not be null");
+
+        if (model.isEmpty()) {
+            log.debug("addData called with an empty model – nothing to do.");
+            return;
+        }
+
+        StringWriter writer = new StringWriter();
+        model.write(writer, "ttl");
+        String turtle = writer.toString();
+
+        String base = config.getSparqlUpdateEndpoint();      // …/statements
+        URI endpoint;
+        if (StringUtils.isBlank(graphUri)) {
+            endpoint = URI.create(base);                       // default graph
+        } else {
+            String encCtx = URLEncoder.encode("<" + graphUri + ">", StandardCharsets.UTF_8);
+            endpoint = URI.create(base + "?context=" + encCtx); // named graph
+        }
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(endpoint)
+                .header("Content-Type", Lang.TURTLE.getHeaderString()) // text/turtle
+                .POST(HttpRequest.BodyPublishers.ofString(turtle, StandardCharsets.UTF_8))
+                .build();
+
+        try {
```

Comment:
There are significant common parts to replaceGraph and updateGraph - would be good to have a common method for them.